packages/tracing-processing/src/services/enricherService.ts

```diff
@@ -84,11 +95,13 @@ export function dbServiceBuilder(db: DB) {
               );
 
               if (!tenantEservice) {
-                return enrichPurposeWithError(
-                  record,
-                  tracing,
-                  PurposeErrorCodes.ESERVICE_NOT_ASSOCIATED,
-                );
+                return {
+                  purposeId: record.purpose_id,
+                  status: record.status,
+                  errorCode: PurposeErrorCodes.ESERVICE_NOT_ASSOCIATED,
+                  message: PurposeErrorCodes.ESERVICE_NOT_ASSOCIATED,
```

Comment:
once. the check has ben fixed, I suggest this rename
```suggestion
                  errorCode: PurposeErrorCodes.TENANT_IS_NOT_PRODUCER_OR_CONSUMER,
                  message: PurposeErrorCodes.TENANT_IS_NOT_PRODUCER_OR_CONSUMER,
```